### PR TITLE
Add support for empty s3 lifecycle rules

### PR DESCRIPTION
**Pull request by ayablonskyy-amplify**

### CODE REVIEW

File: jenkins/disco_bucket_update.sh [hunk 1]
```diff
@@ -89,9 +89,11 @@ function update_bucket() {
         CANONICAL_ID="$(aws --region $bucket_region s3api get-bucket-acl --bucket $bucket_name | jq -r .Owner.ID)"
     fi
 
-    # Apply the lifecycle policy
-    aws s3api --region $bucket_region put-bucket-lifecycle --bucket $bucket_name \
-        --lifecycle-configuration file://$lifecycle_file
+    if [ -s "$livecycle_file" ] ; then
```
Comment (skim-amplify) [inline jenkins/disco_bucket_update.sh]: typo: `lifecycle_file`

**Comment by pchand-amplify** [comment]
@eugenejen Do you have any concerns with not having a lifecycle policy defined?

**Comment by jconstance-amplify** [comment]
Is this still needed or was this worked around?

**Comment by pchand-amplify** [comment]
👍

**Comment by eugenejen** [comment]
damn. i did not see this PR until it is closed :-( 

@pchand-amplify : iirc, disco had problemd before when bucket got glaciered. especially the one contains important configurations and credentials. I don't know the behavior without lifecycle policy for S3 bucket. But as long as you are sure the change won't caused bucket get glaciers if no updates happens. then you can go ahead.

**Comment by pchand-amplify** [comment]
@eugenejen : Looking that docs it seems like the default behaviour is no archiving or versioning. So the files will not be deleted.

**Pull request merged**